src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -512,7 +518,26 @@ private Map<Expression, NavigableSet<SSTableIndex>> referenceAndGetView(Operatio
         }
         finally
         {
-            Tracing.trace("Querying storage-attached indexes {}", indexNames);
+            traceGroupedIndexes(referencedIndexes, queryContext);
```

Comment:
why is this part (line 530) executed even when tracing is disabled? does it have important side effects?

Reply:
I'll rename the method. It tracks metrics (queryContext::addIndexesQueried) too.